chaintesting/action_test_helpers.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chaintesting
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/state"
+)
+
+var _ state.Mutable = (*inMemoryStore)(nil)
+
+// inMemoryStore is a storage that acts as a wrapper around a map and implements state.Mutable.
+type inMemoryStore struct {
+	Storage map[string][]byte
+}
+
+func NewInMemoryStore() *inMemoryStore {
+	return &inMemoryStore{
+		Storage: make(map[string][]byte),
+	}
+}
+
+func (s *inMemoryStore) GetValue(_ context.Context, key []byte) ([]byte, error) {
+	val, ok := s.Storage[string(key)]
+	if !ok {
+		return nil, database.ErrNotFound
+	}
+	return val, nil
+}
+
+func (s *inMemoryStore) Insert(_ context.Context, key []byte, value []byte) error {
+	s.Storage[string(key)] = value
+	return nil
+}
+
+func (s *inMemoryStore) Remove(_ context.Context, key []byte) error {
+	delete(s.Storage, string(key))
+	return nil
+}
+
+// ActionTest is a single parameterized test. It calls Execute on the action with the passed parameters
+// and checks that all assertions pass.
+type ActionTest struct {
+	Action chain.Action
```

Comment:
I think it's more idiomatic to add a `string` field `Name` here instead of putting the test name as the key in `map[string]ActionTest` in the `Suite` struct below. I think if we do this we can make that map type just a slice.

Reply:
Thanks ! So I have removed this `ActionTestSuite` type and instead passed the `[]ActionTest` as part of the `Run` function.

Reply:
(And it's now passing the `Name` as one of the fields of `ActionTest`.